ibis/backends/tests/test_generic.py

```diff
@@ -53,13 +53,32 @@
 @pytest.mark.notyet(["flink"], "The runtime does not support untyped `NULL` values.")
 def test_null_literal(con, backend):
     expr = ibis.null()
-    result = con.execute(expr)
-    assert pd.isna(result)
+    assert pd.isna(con.execute(expr))
 
     with contextlib.suppress(com.OperationNotDefinedError):
         backend_name = backend.name()
         assert con.execute(expr.typeof()) == NULL_BACKEND_TYPES[backend_name]
 
+    with pytest.raises(AttributeError):
+        expr.upper()
+    with pytest.raises(AttributeError):
+        expr.cast(str).max()
```

Comment:
These tests that are testing the lack of a particular method aren't that useful and add visual noise to the test suite. Testing that `upper` method doesn't exist on a non-string type is testing that a method named upper was not defined. We just don't define the method and are there many such non-existent methods across the codebase. I don't think it's a good use of code to test that we don't define this or that method.

Since those methods are entirely determined by the expression's dtype--by design--why not just check that the value of `expr.type()` is something in particular?

`max` on `str` is also likely to break, I suspect we'll probably add that at some point in the near future.

Reply:
> why not just check that the value of expr.type() is something in particular?

much better idea. Changed to that.